spec/src/lib.rs

```diff
@@ -282,40 +281,33 @@ impl Genesis {
             })
         }
 
-        fn build_dep_group_output(
-            out_points: Vec<packed::OutPoint>,
-            lock: packed::Script,
-        ) -> Result<(packed::CellOutput, Bytes), Box<dyn Error>> {
-            let data = Bytes::from(out_points.pack().as_slice());
-            let cell = packed::CellOutput::new_builder()
-                .data_hash(packed::CellOutput::calc_data_hash(&data).pack())
-                .lock(lock)
-                .build_exact_capacity(Capacity::bytes(data.len())?)?;
-            Ok((cell, data))
-        }
-
-        let secp_data_out_point =
-            find_out_point_by_data_hash(cellbase_tx, &CODE_HASH_SECP256K1_DATA)
-                .expect("Get secp data out point failed");
-        let secp_blake160_out_point =
-            find_out_point_by_data_hash(cellbase_tx, &CODE_HASH_SECP256K1_BLAKE160_SIGHASH_ALL)
-                .expect("Get secp blake160 out point failed");
-        let secp_ripemd160_out_point = find_out_point_by_data_hash(
-            cellbase_tx,
-            &CODE_HASH_SECP256K1_RIPEMD160_SHA256_SIGHASH_ALL,
-        )
-        .expect("Get secp ripemd160 out point failed");
-
-        let (cell_blake160, data_blake160) = build_dep_group_output(
-            vec![secp_data_out_point.clone(), secp_blake160_out_point.clone()],
-            self.system_cells.lock.clone().into(),
-        )?;
-        let (cell_ripemd160, data_ripemd160) = build_dep_group_output(
-            vec![secp_data_out_point.clone(), secp_ripemd160_out_point],
-            self.system_cells.lock.clone().into(),
-        )?;
-        let outputs = vec![cell_blake160, cell_ripemd160];
-        let outputs_data = vec![data_blake160.pack(), data_ripemd160.pack()];
+        let (outputs, outputs_data): (Vec<_>, Vec<_>) = self
+            .dep_groups
+            .values()
+            .map(|files| {
+                let out_points: Vec<_> = files
+                    .iter()
+                    .map(|res| {
+                        let data: Bytes = res.get()?.into_owned().into();
```

Comment:
I think it is a bit overhead to read the data to just check what's the position of it in the cellbase. We also can search the res in the system cells files list. Yes, it means the user must reference the same resource via the same path, but it is acceptable.

Reply:
Or, we can cache the hash result?
`HashMap<Resource, H256>`